include/SFML/Graphics/Rect.hpp

```diff
@@ -79,59 +70,124 @@ class Rect
+    ///
+    ////////////////////////////////////////////////////////////
+    constexpr Vector2<T> getStart() const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Get the bottom-right point of the rectangle
+    ///
+    /// \return Bottom-right point of the rectangle
+    ///
+    ////////////////////////////////////////////////////////////
+    constexpr Vector2<T> getEnd() const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Get the center point of the rectangle
+    ///
+    /// \return Center point of the rectangle
+    ///
+    ////////////////////////////////////////////////////////////
+    constexpr Vector2<T> getCenter() const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Get the absolute size of the rectangle
+    ///
+    /// \return Absolute size of the rectangle
+    ///
+    ////////////////////////////////////////////////////////////
+    constexpr Vector2<T> getAbsoluteSize() const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Get positive sized representaion of the rectangle
+    ///
+    /// \return Same rectangle, represented with positive size
     ///
     ////////////////////////////////////////////////////////////
-    constexpr std::optional<Rect<T>> findIntersection(const Rect<T>& rectangle) const;
+    constexpr Rect<T> getAbsoluteRect() const;
 
     ////////////////////////////////////////////////////////////
-    /// \brief Get the position of the rectangle's top-left corner
+    /// \brief Check if a point is inside the rectangle's area
+    ///
+    /// This check includes the left and top edges of the rectangle
+    /// as it's area, but excludes right and bottom edges from it.
     ///
-    /// \return Position of rectangle
+    /// \param point Point to check
     ///
-    /// \see getSize
+    /// \return True if the point is inside the rectangle, false otherwise
+    ///
+    /// \see findIntersection
     ///
     ////////////////////////////////////////////////////////////
-    constexpr Vector2<T> getPosition() const;
+    constexpr bool contains(const Vector2<T>& point) const;
 
     ////////////////////////////////////////////////////////////
-    /// \brief Get the size of the rectangle
+    /// \brief Check if another rectangle is inside this rectangle's area
     ///
-    /// \return Size of rectangle
+    /// This check includes the left and top edges of the rectangle
+    /// as it's area, but excludes right and bottom edges from it.
     ///
-    /// \see getPosition
+    /// \param rectangle Rectangle to check
+    ///
+    /// \return True if the rectangle is inside this rectangle, false otherwise
+    ///
+    /// \see findIntersection
     ///
     ////////////////////////////////////////////////////////////
-    constexpr Vector2<T> getSize() const;
+    constexpr bool contains(const Rect<T>& rectangle) const;
```

Comment:
I'm personally fine with this new API seeing where it is already being used in this PR, but over in the SFML Discord there is no consensus on what its edge case behavior should be so for the purposes of avoiding that discussion halting this PR, perhaps it's easier to remove this new member function and save that conversation for a later issue or PR.

Reply:
In my opinion, this is just a way to evade and postpone the problem. We can add the method with whatever behavior we decide for now, and if we then change our mind, just open another PR. I think this only seems natural to have this method, and since it's new, nobody's gonna be hurt if it's a little imperfect anyway

Reply:
> this is just a way to evade and postpone the problem

Yes, that is my explicit goal. Evading this problem may be more pragmatic than halting this PR until everyone can agree on a solution. 

> We can add the method with whatever behavior we decide for now,

That's what I'm saying. I don't think we'll come to a tentative agreement.

> I think this only seems natural to have this method, and since it's new, nobody's gonna be hurt if it's a little imperfect anyway

Precedent shows that the maintainers don't work this way and won't be eager to merge code that we know to be lacking in quality or rigor. If you insist on keeping this API in the PR, expect it to prolong the review time.

Reply:
You've got a point